File: internal/content/testdata/bench/typescript/nc-rotting-oranges.ts
function oranges_rotting(grid: number[][]): number {
    const m = grid.length;
    const n = grid[0].length;
    const queue: [number, number, number][] = [];
    let fresh = 0;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (grid[i][j] === 2) queue.push([i, j, 0]);
            else if (grid[i][j] === 1) fresh++;
        }
    }
    if (fresh === 0) return 0;
    let minutes = 0;
    let qi = 0;
    while (qi < queue.length) {
        const [i, j, t] = queue[qi++];
        for (const [di, dj] of [[1,0],[-1,0],[0,1],[0,-1]]) {
            const ni = i + di, nj = j + dj;
            if (ni >= 0 && ni < m && nj >= 0 && nj < n && grid[ni][nj] === 1) {
                grid[ni][nj] = 2;
                fresh--;
                minutes = t + 1;
                queue.push([ni, nj, t + 1]);
            }
        }
    }
    return fresh === 0 ? minutes : -1;
}

File: internal/content/testdata/bench/csharp/nc-rotting-oranges.cs
using System.Collections.Generic;

public class Solution {
    public long oranges_rotting(long[][] grid) {
        int m = grid.Length, n = grid[0].Length;
        var queue = new Queue<int[]>();
        long fresh = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] == 2) queue.Enqueue(new int[]{i, j, 0});
                else if (grid[i][j] == 1) fresh++;
            }
        }
        if (fresh == 0) return 0;
        long minutes = 0;
        int[][] dirs = new int[][]{ new int[]{1,0}, new int[]{-1,0}, new int[]{0,1}, new int[]{0,-1} };
        while (queue.Count > 0) {
            int[] cur = queue.Dequeue();
            int i = cur[0], j = cur[1], t = cur[2];
            foreach (int[] d in dirs) {
                int ni = i + d[0], nj = j + d[1];
                if (ni >= 0 && ni < m && nj >= 0 && nj < n && grid[ni][nj] == 1) {
                    grid[ni][nj] = 2;
                    fresh--;
                    minutes = t + 1;
                    queue.Enqueue(new int[]{ni, nj, t+1});
                }
            }
        }
        return fresh == 0 ? minutes : -1;
    }
}
